x-pack/test/plugin_api_integration/test_suites/task_manager/health_route.ts

```diff
@@ -96,6 +98,21 @@ export default function ({ getService }: FtrProviderContext) {
       .then((response) => response.body);
   }
 
+  function getHealthForSampleTask(): Promise<MonitoringStats> {
+    return retry.try(async () => {
+      const health = await getHealth();
+
+      // only return health stats once they contain sampleTask, if requested
+      if (health.stats.runtime.value.drift_by_type.sampleTask) {
+        return health;
+      }
+
+      // if sampleTask is not in the metrics, wait a bit and retry
+      delay(500);
```

Comment:
Does this work? I am not sure if it works without an await. i think it's just pushed to event loop.

Reply:
good catch, yes, it's basically a no-op - will need to add an `await` ...

Reply:
added an `await` in  commit 04cb7938b970e6fe32ea16092bfdbb4d619586ce